environment.go

```diff
@@ -284,10 +284,24 @@ func (e *envState) initSettings(path string) error {
 }
 
 func (e *envState) initDB(ctx context.Context, settings DBSettings) error {
+
 	opts := options.Client().ApplyURI(settings.Url).SetWriteConcern(settings.WriteConcernSettings.Resolve()).
 		SetReadConcern(settings.ReadConcernSettings.Resolve()).
 		SetConnectTimeout(5 * time.Second).SetMonitor(apm.NewLoggingMonitor(ctx, time.Minute, apm.NewBasicMonitor(nil)).DriverAPM())
 
+	if settings.HasAuth() {
+		ymlUser, ymlPwd, err := settings.GetAuth()
+		if err != nil {
+			return errors.Wrap(err, "problem getting auth from yaml authfile")
```

Comment:
Do we need to return here? Maybe we should try to connect without auth? 
Like, what if the file is present but empty?

Reply:
I think if the auth file is set, we should expect the credentials to work. If the file is present but empty, and we don't need it, the environment variable should be unset to give us the no-auth behavior we want. If we set the auth file and need to authenticate, the file should not be empty.

Reply:
Still, the process's environment is hard to manage. The strategy in cedar resonated with me: if there's a file and if [the file contains parsable credentials](https://github.com/evergreen-ci/cedar/blob/21fcb86a728e54a3bad1b2cf72535f5da01ee58d/operations/setup.go#L167) then [use the credentials](https://github.com/evergreen-ci/cedar/blob/21fcb86a728e54a3bad1b2cf72535f5da01ee58d/environment.go#L61). Otherwise we pass on the credential.

Reply:
following up on this

Reply:
forgot to remove the return